c8/c8ql.py

```diff
@@ -346,3 +348,30 @@ def response_handler(resp):
             return resp.body
 
         return self._execute(request, response_handler)
+
+
+    def get_all_batches(self, query, bind_vars=None, batch_size=1000):
```

Comment:
@amit-macrometa @ivayloPetkov please review this function for any possible issues with DB. Thanks!

Reply:
We are doing the same thing here.. we create the cursor API and then call the PUT /api/cursor/{cursor-id} ..
It's just that instead of user doing this everytime we are doing it ourselves.. we have observed that users try to fetch all docs in a collection or any query with filter and they complain they get 1000 docs. This is a common use case and ideally our sdks should have a common function to enhance their experience.

Reply:
Even Durga had complained about the same issue:
`So went to pyC8 github repo to check out are there any examples in it Readme.md file. Cut-n-pasted from one of the code examples on the Readme.md file. Now I got my program to read from a collection but it was returning only 1000 documents. I need to get more docs.`

Reply:
I understand @koshy-thomas that the new function is doing what user of the APIs is supposed to do.

My concern is, with your function, we would be returning millions of records in the `result` of this new function. This is my concern here. Should we be doing this to make life of the user easier !!!

Whatever we decide, it should be a conscious decision. My view is unless we have specific request from users (many users infact), we can avoid this change and just update our documentation with examples on how to retrieve all documents - in batches.

@ivayloPetkov is it fine to return all documents ?

Reply:
@amit-macrometa what should the user do in case he wants to fetch 1 million docs? Does his approach change?
Are we still going to still recommend him the same approach of `creating cursor `and calling `read next batch` API ? If yes, then I don't see a reason why we shouldn't have a common function (having a common function to `append data and return results` should not create the problem ) (It's the approach to the requirement that can cause any problem). So that's why I want to know what should be the recommended approach in terms of API if a user wants to fetch more than 1 million docs?

Reply:
I got your point @koshy-thomas . But recently we had a similar discussion happening on Slack regarding GUI (which is going to be run on client's browser) to handle `how many documents` . I am coming from there.

https://macrometa-team.slack.com/archives/C02S13S2R9T/p1667201684382389